test/test_binarytree.rb

```diff
@@ -302,17 +302,21 @@ def test_old_camelCase_method_names
       @left_child2  = Tree::BinaryTreeNode.new("A Child at Left", "Child Node @ left")
       @right_child2 = Tree::BinaryTreeNode.new("B Child at Right", "Child Node @ right")
 
-      require 'structured_warnings'
-
       meth_names_for_test = %w{leftChild isLeftChild? rightChild isRightChild?}
 
       meth_names_for_test.each do |meth_name|
-        assert_warn(DeprecatedMethodWarning) {@root.send(meth_name)}
+        output_warning  = capture_output do
+          @root.send(meth_name)
+        end
+        assert_not_empty(output_warning.last)
       end
 
-      assert_warn(DeprecatedMethodWarning) {@root.leftChild = @left_child2}
-      assert_warn(DeprecatedMethodWarning) {@root.rightChild = @right_child2}
-      assert_raise(NoMethodError) {@root.to_snake_case("ABCD")} # Make sure the right method is visible
+      assert_not_empty(capture_output{@root.leftChild = @left_child2}.last)
+      assert_not_empty(capture_output{@root.rightChild = @right_child2}.last)
+      #ruby self include private policy changed in 2.3
+      if RUBY_VERSION<'2.3.0'
```

Comment:
IMO this test shows the presence of an issue in this library in Ruby >= 2.3.
The bug is that `to_snake_case` method that was intended to be private is indeed public because of a behavior change in Ruby.
So a proper fix would arguably be to make `to_snake_case` method private. It seems `method_missing` and `to_snake_case` can be just made not nested methods (i.e. `def self.included` can be removed).

By the way, I filed an issue for Ruby here - https://bugs.ruby-lang.org/issues/13249